### PR TITLE
CACHEPOT_BASEDIR

### PR DESCRIPTION
**Pull request by montekki**

### CODE REVIEW

**Comment by montekki** [comment]
@drahnr, marking this ready for review.

Just strip `CARGO_MANIFEST_DIR` from the paths, dont cache the `cwd` and call it a day. The testing i've done shows that it improves the cache hit rate somewhat but quite honestly I do not think that the task at hand is doable cleanly and safely for all usecases at this point.

Testing done: build `cachepot` with `cachepot`.

There still are a lot of absulute paths that end up in the build artifacts such as
1) paths in *.d files.
2) paths in artifacts from the `cc` builds
3) paths in things that are still built locally such as `cdylib` crates.

File: src/compiler/rust.rs [hunk 1]
```diff
@@ -1453,8 +1454,28 @@ where
                 val.hash(&mut HashToDigest { digest: &mut m });
             }
         }
+        let mut arguments = arguments;
         // 8. The cwd of the compile. This will wind up in the rlib.
-        cwd.hash(&mut HashToDigest { digest: &mut m });
+        // This will end up in the rlib only in the case no basedir path
+        // rewriting is done and the build is operating with the absolute paths.
+        if let Ok(_idx) =
+            env_vars.binary_search_by(|(ref k, _)| k.cmp(&OsString::from("CACHEPOT_REMAP_RELATIVE")))
```
Comment (drahnr) [inline src/compiler/rust.rs:1462]: I would assume we want this to be on by default?

**Review by drahnr — APPROVED** [review]

File: src/compiler/rust.rs (hunk 1 above)
Comment (montekki) [reply src/compiler/rust.rs:1462]: Maybe roll this out without being default for everyone and verify it on some lower-risk scenarios such as maybe CI targets that do not go into releases and such?

**Comment by drahnr** [comment]
A rebase and this should be good to go :)

**Comment by drahnr** [comment]
`rustfmt` seems unhappy, I'd like to land this soon™ :)

**Pull request merged**